### PR TITLE
DOC: shift emphasis in known issues, clarify that linspace works

### PR DESCRIPTION
**Pull request by lpsinger**
### Description

Update documentation on array creation routines and units: `numpy.linspace` is known to work with quantity input.

- [ ] By checking this box, the PR author has requested that maintainers do **NOT** use the "Squash and Merge" button. Maintainers should respect this when possible; however, the final decision is at the discretion of the maintainer that merges the PR.

### CODE REVIEW

File: docs/changes/16989.other.rst
Comment (pllim) [inline docs/changes/16989.other.rst:1]: Usually doc change does not need change log but I'll defer to Marten. Thanks!

File: docs/known_issues.rst
Comment (pllim) [inline docs/known_issues.rst:1]: Do we have a test (not just doctest) to demonstrate that it really is working? If not, such a test should be added.

Comment (mhvk) [reply docs/known_issues.rst:1]: Yes: https://github.com/astropy/astropy/blob/05bb236a8cd61c127987ff6de8db92ac3cedf612/astropy/units/tests/test_quantity_non_ufuncs.py#L1275-L1285

File: docs/known_issues.rst [hunk 1]
```diff
@@ -146,11 +146,6 @@ Workarounds include moving the units outside of the call to
     >>> np.arange(0, 10, 1) * u.m
     <Quantity [0., 1., 2., 3., 4., 5., 6., 7., 8., 9.] m>
 
-Also, `~numpy.linspace` does work:
```
Comment (mhvk) [inline docs/known_issues.rst:149]: I'm confused - why do we want to remove this? It is to give an alternative to using `np.arange`, which does not work with units.

**Review by mhvk — CHANGES_REQUESTED** [review]
I'm confused - see in-line comment...

File: docs/known_issues.rst (hunk 1 above)
Comment (mhvk) [reply docs/known_issues.rst:149]: Sounds good! Might as well fix that since it clearly confused you, so will likely confuse others...

**Review by pllim — COMMENTED** [review]
LGTM but I'll leave the approval for Marten. Thanks!

**Review by mhvk — APPROVED** [review]
OK, thanks, let's get this in!

**Pull request merged**